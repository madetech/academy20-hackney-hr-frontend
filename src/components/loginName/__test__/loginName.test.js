import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import LoginName from "../loginName.js";

let container = null;
beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null
});

it("renders with or without a name", () => {
    act(() => {
        render(<LoginName/>, container);
    });
    expect(container.textContent).toBe("Hello, Employee");
    act(() => {
        render(<LoginName name="Bella Cockrell"/>, container);
    });
    expect(container.textContent).toBe("Bella Cockrell");
});

