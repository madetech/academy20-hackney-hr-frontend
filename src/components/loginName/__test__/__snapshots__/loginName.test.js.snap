// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name is rendered in header 1`] = `
<h1>
  Hello, Employee
</h1>
`;
